lib/syncthing/utils.go

```diff
@@ -178,7 +178,11 @@ func TryMigrateDatabase(ctx context.Context, deleteRetention time.Duration, apiA
 		api := migratingAPI{addr: apiAddr}
 		apiCtx, cancel := context.WithCancel(ctx)
 		defer cancel()
-		go api.Serve(apiCtx)
+		go func() {
+			if err := api.Serve(apiCtx); err != nil {
+				slog.Warn("Failed to serve temporary API server", slogutil.Error(err))
```

Comment:
Ah, unfortunately, the better way is like it was despite what the linter thinks. We always get a non-nil error here since we shut down the server when we're done with it, and it unnecessary effort (imho) to try to distinguish the intentional shutdown from another problem.

Reply:
OK